### PR TITLE
VDS-1763: Preview of contacts import

### PR DESCRIPTION
**Pull request by asvishnyakov**
Description: 

--

QA-test: VDS-1295

Demo-test: VDS-1297

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.CustomerExportImport_1.0.0-pr-14-d6821a0c.zip

### CODE REVIEW

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        public string AccountStatus { get; set; }
+
+        [Name("Email Verified")]
+        public bool? EmailVerified { get; set; }
+
+        [Name("Contact Status")]
+        public string ContactStatus { get; set; }
+
+        [Name("Associated Organization Id")]
+        public string AssociatedOrganizationId { get; set; }
```
Comment (trueboroda) [inline src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs:65]: Ids.

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        public Contact ToContact()
+        {
+            return new Contact()
+            {
+                Id = Id,
+                FirstName = FirstName,
+                LastName = LastName,
+                FullName = FullName,
+                OuterId = ContactOuterId,
+                SecurityAccounts = new List<ApplicationUser>()
```
Comment (trueboroda) [inline src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs:100]: What if accountId is null?

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                        StoreId = StoreId,
+                        UserName = AccountLogin,
+                        Email = AccountEmail,
+                        UserType = AccountType,
+                        Status = AccountStatus,
+                        EmailConfirmed = EmailVerified ?? false
+                    }
+                },
+                Status = ContactStatus,
+                AssociatedOrganizations = new List<string>() { AssociatedOrganizationId },
```
Comment (trueboroda) [inline src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs:114]: Ids

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                BirthDate = BirthDate,
+                TimeZone = TimeZone,
+                Phones = Phones?.Split(","),
+                Groups = UserGroups?.Split(","),
+                Addresses = new List<Address>()
```
Comment (trueboroda) [inline src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs:119]: What if there is now addresses?

File: src/VirtoCommerce.CustomerExportImportModule.Web/Controllers/ImportController.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    public class ImportController : ControllerBase
+    {
+        private readonly IBlobStorageProvider _blobStorageProvider;
+        private readonly ICustomerImportPagedDataSourceFactory _csvPagedPriceDataSourceFactory;
+
+        public ImportController(IBlobStorageProvider blobStorageProvider,
+            ICustomerImportPagedDataSourceFactory csvPagedPriceDataSourceFactory)
+        {
+            _blobStorageProvider = blobStorageProvider;
+            _csvPagedPriceDataSourceFactory = csvPagedPriceDataSourceFactory;
```
Comment (trueboroda) [inline src/VirtoCommerce.CustomerExportImportModule.Web/Controllers/ImportController.cs:24]: Rename csvPagedPriceDataSourceFactory

**Review by trueboroda — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs (hunk 1 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs]: I refactored code so now both export & import will share the same model

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs (hunk 2 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs]: Fixed, but I think we need general solution to mark some fields nullable or not

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs (hunk 3 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs]: I refactored code so now both export & import will share the same model

File: src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs (hunk 4 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerExportImportModule.Core/Models/ImportableContact.cs]: It should be resolved in data source, but thank you, that's important reminder

File: src/VirtoCommerce.CustomerExportImportModule.Web/Controllers/ImportController.cs (hunk 5 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerExportImportModule.Web/Controllers/ImportController.cs]: Fixed

**Review by ilyawzrd — APPROVED** [review]

**Review by pushnitsa — APPROVED** [review]

**Review by trueboroda — APPROVED** [review]

**Pull request merged**